homeworks/edgar_meloyan/Test/TriangleDemo.java

```diff
@@ -0,0 +1,75 @@
+import java.util.Scanner;
+
+public class TriangleDemo {
+    public static void main(String[] args) {
+        Scanner console = new Scanner(System.in);
+
+        System.out.println("Enter AB coordinates: ");
+        double x1 = console.nextDouble();
+        double y1 = console.nextDouble();
+
+        System.out.println("Enter BC coordinates: ");
+        double x2 = console.nextDouble();
+        double y2 = console.nextDouble();
+
+        System.out.println("Enter CA coordinates: ");
+        double x3 = console.nextDouble();
+        double y3 = console.nextDouble();
+
+        Triangle obj1 = new Triangle(x1, y1, x2, y2, x3, y3);
+        System.out.println(obj1.areOfTriangle());
+        System.out.println(obj1.perimeterOfTriangle());
+    }
+}
+
+class Triangle {
+    Point point1;
+    Point point2;
+    Point point3;
+
+    Triangle(double x1, double y1, double x2, double y2, double x3, double y3) {
+        point1 = new Point(x1, y1);
+        point2 = new Point(x2, y2);
+        point3 = new Point(x3, y3);
+    }
+
+
+    double firstLineLength(Point point1, Point point2) {
+        return Math.sqrt(Math.pow(point2.x - point1.x, point2.x - point1.x) + Math.pow(point2.y - point1.y, point2.y - point1.y));
+    }
+
+    double secondLineLength(Point point3, Point point2) {
+        return Math.sqrt(Math.pow(point3.x - point2.x, point3.x - point2.x) + Math.pow(point3.y - point2.y, point3.y - point2.y));
+    }
+
+    double thirdLineLength(Point point1, Point point3) {
+        return Math.sqrt(Math.pow(point3.x - point1.x, point3.x - point1.x) + Math.pow(point3.y - point1.y, point3.y - point1.y));
+    }
+
+    double perimeterOfTriangle() {
+        return firstLineLength(point1, point2) + secondLineLength(point2, point3) + thirdLineLength(point1, point3);
+    }
+
+    double halfPer() {
+        return perimeterOfTriangle() / 2.0;
+    }
+
+    double areOfTriangle() {
```

Comment:
what was the problem you faced, could you describe a little

Reply:
When I start debugging the firstlength method, IDEA offers me to choose to run one of the nested methods: maths.sqrt or math.pow. After I select one of these methods, a message pops up saying that the sqrt or pow method has not been called. And then, the debugger jumps to the next second length method.